rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -125,26 +136,63 @@ get_node_names(
       RMW_SET_ERROR_MSG("could not allocate memory for node namespace");
       goto fail;
     }
+
+    ++named_nodes_num;
+  }
+
+  // prepare actual output without empty names
+  // TODO: implement functionality that allocates just enough memory for the output buffer,
+  // so using additional buffer would be unnessecary 
+  rcutils_ret = rcutils_string_array_init(node_names, named_nodes_num, &allocator);
+  if (rcutils_ret != RCUTILS_RET_OK) {
+    RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
+    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+  }
+
+  for (auto i = 0; i < named_nodes_num; ++i) {
+    node_names->data[i] = rcutils_strdup(tmp_names_list.data[i], allocator);
```

Comment:
This logic looks incorrect. It's possible there is an empty name in the middle of `tmp_names_list`, in which case we're copying `nullptr` (and missing a potential valid name at the end of the list).

The for-loop should loop up to `length` and then inside we should guard against `nullptr`:

```c++
size_t j = 0;
for (auto i = 0; i < length; ++i) {
  if (tmp_names_list.data[i]) {
    node_names->data[j] = rcutils_strdup(tmp_names_list.data[i], allocator);
  }
  ++j;
}
```

Reply:
I overlooked the logic above (https://github.com/ros2/rmw_connext/pull/362/files#r327540360), this loop looks okay.